vsphere_storage_plugin/cidata.py

```diff
@@ -91,3 +91,32 @@ def delete(rawvolume_client, **kwargs):
     rawvolume_client.delete_file(datacenter_id=datacenter_id,
                                  datacenter_name=datacenter_name,
                                  datastorepath=storage_path)
+
+
+@op
+@with_rawvolume_client
+def upload_iso(rawvolume_client,
+               datacenter_name,
+               allowed_datastores,
+               allowed_datastore_ids,
+               volume_prefix,
+               iso_file_path,
+               **_):
+    is_node_deprecated(ctx.node.type)
+    if ctx.instance.runtime_properties.get(VSPHERE_STORAGE_FILE_NAME):
```

Comment:
Is this runtime property cleaned up on uninstall? For example, maybe install failed. They ran rollback. But the property is already set, but we do not actually have the instance.